devito/ir/iet/nodes.py

```diff
@@ -228,6 +229,10 @@ def free_symbols(self):
     def defines(self):
         return ()
 
+    @property
+    def children(self):
```

Comment:
this is a bit hacky as it's bypassing a class attribute?

Reply:
What would be a better way?

Reply:
don't know...off top of my head (pseudocode)

```
class Call(...):
    ...
    children = ('nested',)

    @cached_property
     def nested(self):
           return tuple(i for i in self.arguments if isinstance(i, Call))
```
?

Reply:
the problem with that is that `_traversable` needs to have `__init__` arguments (so I can't use a `nested` property, but I can use `arguments`). `arguments` seems to work though. (although I can't find any logic to skip non-Node objects?)

Reply:
sorry, my bad, that's what I meant! ``_traversable``, not ``children``

what error do you get with ``_traversable`` in place of ``children``?

Reply:
with `_traversable = ['arguments']`, none.
with `_traversable = ['nested'`]` (with nested as per your code) I get 
```
def _rebuild(self, *args, **kwargs):
        """Reconstruct self."""
        handle = self._args.copy()  # Original constructor arguments
        argnames = [i for i in self._traversable if i not in kwargs]
        handle.update(OrderedDict([(k, v) for k, v in zip(argnames, args)]))
        handle.update(kwargs)
>       return type(self)(**handle)
E       TypeError: __init__() got an unexpected keyword argument 'nested'
```